### PR TITLE
Electron-118 (Screen snippet always on top)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
Implemented a method to turn off `alwaysOnTop` while using screen snippet (windows).

@VikasShashidhar , @VishwasShashidhar & @lneir Please review. Thanks.

### CODE REVIEW

File: js/screenSnippet/index.js [hunk 1]
```diff
@@ -61,6 +63,17 @@ class ScreenSnippet {
                     captureUtil = path.join(execPath, 'ScreenSnippet.exe');
                 }
 
+                // Method to verify and disable always on top property
+                // as an issue with the ScreenSnippet.exe not being on top
+                // of the electron wrapper
+                const windows = electron.BrowserWindow.getAllWindows();
+                if (windows && windows.length > 0) {
+                    isAlwaysOnTop = windows[ 0 ].isAlwaysOnTop();
```
Comment (KiranNiranjan) [inline js/screenSnippet/index.js]: Is it fine if i read this from electron api or should i read it from config file?

@VikasShashidhar, @VishwasShashidhar & @lneir

Comment (VishwasShashidhar) [reply js/screenSnippet/index.js]: Either way should be fine. I'd prefer using the electron api as it is dynamic.

File: js/screenSnippet/index.js [hunk 2]
```diff
@@ -74,9 +87,17 @@ class ScreenSnippet {
...
                 if (error && error.killed) {
+                    // Method to reset always on top feature
+                    if (isAlwaysOnTop) {
+                        eventEmitter.emit('isAlwaysOnTop', true);
+                    }
                     // processs was killed, just resolve with no data.
                     resolve();
                 } else {
+                    // Method to reset always on top feature
+                    if (isAlwaysOnTop) {
```
Comment (VishwasShashidhar) [inline js/screenSnippet/index.js]: Can we eliminate duplicate code for the if condition and move it after the `if-else` block?

File: js/screenSnippet/index.js [hunk 3]
```diff
@@ -61,6 +63,17 @@ class ScreenSnippet {
                     captureUtil = path.join(execPath, 'ScreenSnippet.exe');
                 }
 
+                // Method to verify and disable always on top property
+                // as an issue with the ScreenSnippet.exe not being on top
+                // of the electron wrapper
+                const windows = electron.BrowserWindow.getAllWindows();
```
Comment (lneir) [inline js/screenSnippet/index.js]: this looks like it is only is changing isAlwaysOnTop value for first windows (which i would assume is main window) but have you tried this for the pop-outs.  i would assume same issue will happen there.

Comment (KiranNiranjan) [reply js/screenSnippet/index.js]: I'm just reading the value here. If `alwaysOnTop` is enabled i'm using the `eventEmitter` to `setAlwaysOnTop` for all the window in `windowMgr.js:501`

Comment (lneir) [reply js/screenSnippet/index.js]: yes but main question: what happens when always on top is enable and you take screensnippet from a pop-out.  does the issue happen there as well?  or this only an issue for the main window?

Comment (KiranNiranjan) [reply js/screenSnippet/index.js]: Yes this issue happens both on pop-out and main window.

Comment (lneir) [reply js/screenSnippet/index.js]: ok i see, lgtm.

**Pull request merged**